lib/_http_server.js

```diff
@@ -582,10 +581,11 @@ function unconsume(parser, socket) {
   }
 }
 
+const SocketOn = net.Socket.prototype.on;
 function socketOnWrap(ev, fn) {
-  var res = net.Socket.prototype.on.call(this, ev, fn);
-  if (!this.parser) {
-    this.on = net.Socket.prototype.on;
+  const res = SocketOn.call(this, ev, fn);
+  if (this.parser === null) {
+    this.on = SocketOn;
```

Comment:
I'm not so sure it would really help. Making a "fast object" has noticeable overhead and is really only worthwhile for objects that are going to exist for a very long time (e.g. objects created once during startup).